bcipy/acquisition/devices.py

```diff
@@ -0,0 +1,107 @@
+            ex. 'Wearable Sensing DSI-24 dry electrode EEG headset'
+    """
+
+    def __init__(self,
+                 name: str,
+                 channels: List[str],
+                 sample_rate: float,
+                 content_type: str = 'EEG',
+                 connection_methods: List[ConnectionMethod] = None,
+                 description: str = None):
+        self.name = name
+        self.channels = channels
+        self.sample_rate = sample_rate
+        self.content_type = content_type
+        self.connection_methods = connection_methods or [ConnectionMethod.LSL]
+        self.description = description or name
+
+    @property
+    def channel_count(self) -> int:
+        return len(self.channels)
+
+    def analysis_channels(self, exclude_trg: bool = True) -> List[str]:
+        """List of channels used for analysis by the signal module.
+        Parameters:
+        -----------
+            exclude_trg - indicates whether or not to exclude a TRG channel if present.
+        """
+        if exclude_trg:
+            return list(filter(lambda channel: channel != 'TRG',
+                               self.channels))
+        return self.channels
+
+
+def make_device_spec(config: dict) -> DeviceSpec:
+    """Constructs a DeviceSpec from a dict. Throws a KeyError if any fields
+    are missing."""
+    connection_methods = list(
+        map(ConnectionMethod.by_name, config.get('connection_methods', [])))
+    return DeviceSpec(name=config['name'],
+                      content_type=config['content_type'],
+                      channels=config['channels'],
+                      connection_methods=connection_methods,
+                      sample_rate=config['sample_rate'],
+                      description=config['description'])
+
+
+SUPPORTED_DEVICES = {}
+
+
+def load(config_path: Path = Path(DEFAULT_CONFIG)) -> Dict[str, DeviceSpec]:
+    """Load the list of supported hardware for data acquisition from the given
+    configuration file."""
+    global SUPPORTED_DEVICES
+    with open(config_path, 'r', encoding='utf-8') as json_file:
+        specs = [make_device_spec(entry) for entry in json.load(json_file)]
+        SUPPORTED_DEVICES = {spec.name: spec for spec in specs}
+
+
+def supported_devices() -> Dict[str, DeviceSpec]:
+    """Returns the currently supported devices keyed by name."""
+    global SUPPORTED_DEVICES
+    return SUPPORTED_DEVICES
+
+
+def supported_device(name: str) -> DeviceSpec:
+    """Retrieve the DeviceSpec with the given name. An exception is raised
+    if the device is not supported."""
+    global SUPPORTED_DEVICES
+    device = SUPPORTED_DEVICES.get(name, None)
+    if not device:
+        raise ValueError(f"Device not supported: {name}")
+    return device
+
+
+def register(device_spec: DeviceSpec):
+    """Register the given DeviceSpec."""
+    global SUPPORTED_DEVICES
+    SUPPORTED_DEVICES[device_spec.name] = device_spec
+
+
+# Initialize the SUPPORTED_DEVICES
+load()
```

Comment:
Could this be called in init_acquistion or in some other fashion?

Reply:
I'll re-work this a bit to call `load` when SUPPORTED_DEVICES is first accessed. There is other code that uses this module, so I don't want to make that code always have to explicitly load devices, but I agree that this doesn't need to be done at import time.

Rather than module functions, I could refactor this functionality into a class, in which case the load would happen in the `__init__` method.